ens/main.py

```diff
@@ -29,9 +44,30 @@
     raw_name_to_hash,
 )
 
+if TYPE_CHECKING:
+    from web3 import Web3  # noqa: F401
+    from web3.contract import (  # noqa: F401
+        Contract,
+    )
+    from web3.providers import (  # noqa: F401
+        BaseProvider,
+    )
+
+
 ENS_MAINNET_ADDR = '0x314159265dD8dbb310642f98f50C066173C1259b'
 
 
+TxDict = TypedDict("TxDict", {
+    "nonce": int,
+    "gasPrice": int,
+    "gas": int,
+    "from": Union[Address, str],
```

Comment:
I think all of these need to be `AnyAddress`

Reply:
@pipermerriam I'm having trouble using `AnyAddress` though I agree it should be the type here. It might be my understanding of how to use a `TypeVar` mypy throws these errors whenever I try to use `AnyAddress`

> ens/main.py:64: error: Type variable "eth_typing.evm.AnyAddress" is unbound
ens/main.py:64: note: (Hint: Use "Generic[AnyAddress]" or "Protocol[AnyAddress]" base class to bind "AnyAddress" inside a class)
ens/main.py:64: note: (Hint: Use "AnyAddress" in function signature to bind "AnyAddress" inside a function)

I've tried adding a bound to the definition in `eth_typing` but mypy doesn't allow defining both values and a bound in a `TypeVar`

Also, mypy has trouble actually comparing a `AnyAddress` to an `Address`, `ChecksumAddress`, `HexAddress` - and complains (i.e. when the type is defined as `AnyAddress`

> ens/main.py:236: error: Argument 1 to "address_to_reverse_domain" has incompatible type "HexAddress"; expected "Address"